examples/src/main/java/io/dapr/examples/workflows/README.md

```diff
@@ -44,38 +44,434 @@ cd examples
+  /**
+   * The main method of this app.
+   *
+   * @param args The port the app will listen on.
+   * @throws Exception An Exception.
+   */
   public static void main(String[] args) throws Exception {
-    // Register the Workflow with the runtime.
-    WorkflowRuntime.getInstance().registerWorkflow(DemoWorkflow.class);
-    System.out.println("Start workflow runtime");
-    WorkflowRuntime.getInstance().startAndBlock();
+    // Register the Workflow with the builder.
+    WorkflowRuntimeBuilder builder = new WorkflowRuntimeBuilder().registerWorkflow(DemoChainWorkflow.class);
+    builder.registerActivity(ToUpperCaseActivity.class);
+
+    // Build and then start the workflow runtime pulling and executing tasks
+    try (WorkflowRuntime runtime = builder.build()) {
+      System.out.println("Start workflow runtime");
+      runtime.start();
+    }
+
     System.exit(0);
   }
 }
 ```
 
-This application uses `WorkflowRuntime.getInstance().registerWorkflow()` in order to register `DemoWorkflow` as a Workflow in the Dapr Workflow runtime.
+The second Java class you want to look at is `DemoChainWorkflow`, it defines the workflow. In this example it chains the activites in order. See the code snippet below:
+```java
+public class DemoChainWorkflow extends Workflow {
+  @Override
+  public WorkflowStub create() {
+    return ctx -> {
+      ctx.getLogger().info("Starting Workflow: " + ctx.getName());
+
+      String result = "";
+      result += ctx.callActivity(ToUpperCaseActivity.class.getName(), "Tokyo", String.class).await() + ", ";
+      result += ctx.callActivity(ToUpperCaseActivity.class.getName(), "London", String.class).await() + ", ";
+      result += ctx.callActivity(ToUpperCaseActivity.class.getName(), "Seattle", String.class).await();
+
+      ctx.getLogger().info("Workflow finished with result: " + result);
+      ctx.complete(result);
+    };
+  }
+}
+```
+
+The next Java class you want to look at is `ToUpperCaseActivity`, it defines the logics for a single acitvity, in this case, it converts a string to upper case. See the code snippet below:
+```java
+public class ToUpperCaseActivity implements WorkflowActivity {
+
+  @Override
+  public String run(WorkflowActivityContext ctx) {
+    Logger logger = LoggerFactory.getLogger(ToUpperCaseActivity.class);
+    logger.info("Starting Chaining Activity: " + ctx.getName());
+
+    var message = ctx.getInput(String.class);
+    var newMessage = message.toUpperCase();
+
+    logger.info("Message Received from input: " + message);
+    logger.info("Sending message to output: " + newMessage);
 
-The `WorkflowRuntime.getInstance().start()` method will build and start the engine within the Dapr workflow runtime.
+    logger.info("Activity returned: " + newMessage);
+    logger.info("Activity finished");
 
-Now, execute the following script in order to run DemoWorkflowWorker:
+    return newMessage;
+  }
+}
+```
+Execute the following script in order to run DemoChainWorker:
+```sh
```

Comment:
Following the other README files in examples, for auto validation, (`.github/workflows/validate.yaml`), please have the commands enclosed by `dapr/mechanical-markdown` comments such that it is picked up and run automatically.